sdk/go/kubernetes/helm/v2/types.go

```diff
@@ -0,0 +1,195 @@
+type fetchArgs struct {
+	Version     string `pulumi:"version"`
+	CAFile      string `pulumi:"caFile"`
+	CertFile    string `pulumi:"certFile"`
+	KeyFile     string `pulumi:"keyFile"`
+	Destination string `pulumi:"destination"`
+	Keyring     string `pulumi:"keyring"`
+	Password    string `pulumi:"password"`
+	Repo        string `pulumi:"repo"`
+	UntarDir    string `pulumi:"untarDir"`
+	Username    string `pulumi:"username"`
+	Home        string `pulumi:"home"`
+	Devel       *bool   `pulumi:"devel"`
+	Prov        *bool   `pulumi:"prov"`
+	Untar       *bool   `pulumi:"untar"`
+	Verify      *bool   `pulumi:"verify"`
+}
+
+type FetchArgsInput interface {
+	pulumi.Input
+
+	ToFetchArgsOutput() FetchArgsOutput
+	ToFetchArgsOutputWithContext(context.Context) FetchArgsOutput
+}
+
+func (FetchArgs) ElementType() reflect.Type {
+	return reflect.TypeOf((*fetchArgs)(nil)).Elem()
+}
+
+func (i FetchArgs) ToFetchArgsOutput() FetchArgsOutput {
+	return i.ToFetchArgsOutputWithContext(context.Background())
+}
+
+func (i FetchArgs) ToFetchArgsOutputWithContext(ctx context.Context) FetchArgsOutput {
+	return pulumi.ToOutputWithContext(ctx, i).(FetchArgsOutput)
+}
+
+type FetchArgsOutput struct{ *pulumi.OutputState }
+
+func (FetchArgsOutput) ElementType() reflect.Type {
+	return reflect.TypeOf((*fetchArgs)(nil)).Elem()
+}
+
+func (o FetchArgsOutput) ToFetchArgsOutput() FetchArgsOutput {
+	return o
+}
+
+func (o FetchArgsOutput) ToFetchArgsOutputWithContext(ctx context.Context) FetchArgsOutput {
+	return o
+}
+
+// ChartArgs specifies arguments for constructing a Chart resource.
+type ChartArgs struct {
+	// The optional Kubernetes API versions used for Capabilities.APIVersions.
+	ApiVersions pulumi.StringArrayInput
+	// The optional namespace to install chart resources into.
+	Namespace pulumi.StringInput
+	// Overrides for chart values.
+	Values pulumi.MapInput
+	// Transformations is an optional list of transformations to apply to Kubernetes resource definitions
+	// before registering with the engine.
+	Transformations []yaml.Transformation
+	// ResourcePrefix is an optional prefix for the auto-generated resource names. For example, a resource named `bar`
+	// created with resource prefix of `"foo"` would produce a resource named `"foo-bar"`.
+	ResourcePrefix string
+
+	// (Remote chart) The repository name of the chart to deploy. Example: "stable".
+	Repo pulumi.StringInput
+	// (Remote chart) The name of the chart to deploy.  If Repo is specified, this chart name will be prefixed
+	// by the repo name.
+	// Example: Repo: "stable", Chart: "nginx-ingress" -> "stable/nginx-ingress"
+	// Example: Chart: "stable/nginx-ingress" -> "stable/nginx-ingress"
+	Chart pulumi.StringInput
+	// (Remote chart) The version of the chart to deploy. If not provided, the latest version will be deployed.
+	Version pulumi.StringInput
+	// (Remote chart) Additional options to customize the fetching of the Helm chart.
+	FetchArgs FetchArgs
```

Comment:
I think this should be:

```suggestion
	FetchArgs FetchArgsInput
```

The code isn't actually using `FetchArgsInput` currently.